orderer/common/channelparticipation/validator.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package channelparticipation
+
+import (
+	"errors"
+	"fmt"
+
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/channelconfig"
+	"github.com/hyperledger/fabric/protoutil"
+)
+
+// ValidateJoinBlock returns whether this block can be used as a join block for the channel participation API
+// and whether it is an system channel if it contains consortiums, or otherwise
+// and application channel if an application group exists.
+func ValidateJoinBlock(channelID string, configBlock *cb.Block) (isAppChannel bool, err error) {
+	if !protoutil.IsConfigBlock(configBlock) {
+		return false, errors.New("block is not a config block")
+	}
+
+	envelope, err := protoutil.ExtractEnvelope(configBlock, 0)
+	if err != nil {
+		return false, err
+	}
+
+	cryptoProvider := factory.GetDefault()
```

Comment:
unrelated to this PR - I think we should look into why we really need BCCSP just to know whether we have consortium config and application config in the Bundle.